src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFile.cs

```diff
@@ -177,5 +216,14 @@ private static void HashProjectFileDependencyGroups(HashCodeCombiner combiner, I
                 combiner.AddObject(item);
             }
         }
+
+        private static void HashLogMessages(HashCodeCombiner combiner, IList<IAssetsLogMessage> logMessages)
+        {
+            foreach (var item in logMessages.OrderBy(
+                logMessage => @logMessage.Message, StringComparer.OrdinalIgnoreCase))
+            {
+                combiner.AddObject(item);
```

Comment:
I don't know if this will work since AssetsLogMessage doesn't override get hash code, ultimately that is what the hashcodecombiner would use I believe.